loki/transform/dependency_transform.py

```diff
@@ -194,6 +219,39 @@ def rename_imports(self, source, imports, **kwargs):
         if isinstance(source, Subroutine):
             source.body = Transformer(removal_map).visit(source.body)
 
+    def rename_interfaces(self, source, intfs, **kwargs):
+        """
+        Update explicit interfaces to actively transformed subroutines.
+        """
+
+        targets = as_tuple(kwargs.get('targets', None))
+        targets = as_tuple(str(t).upper() for t in targets)
+
+        if self.replace_ignore_items:
+            item = kwargs.get('item', None)
+            targets += as_tuple(str(i).upper() for i in item.ignore) if item else ()
+
+        # Transformer map to remove any outdated interfaces
+        removal_map = {}
+
+        for i in intfs:
+            for b in i.body:
+                if isinstance(b, Subroutine):
+                    target_symbol = b.name.lower()
+
+                    if targets is not None and target_symbol.upper() in targets:
```

Comment:
Better to use the implicit case-insensitive string comparison: `b.name in targets`

Reply:
Yes, but `b.name` is a symbol that has the mixin, and it works in this direction as well:
```python
>>> l = ['a', 'b', 'c']
>>> from loki import Variable
>>> Variable(name='a') in l
True
>>> Variable(name='A') in l
True
>>> Variable(name='aa') in l
False
```

Reply:
Ah yes, my bad, sorry, was suffering from a case of "ramadan brain" there 😆